src/redisearch_rs/c_entrypoint/iterators_ffi/src/metric.rs

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use crate::wrapper::RQEIteratorWrapper;
+use ffi::{
+    IteratorType_METRIC_ITERATOR, QueryIterator, RLookupKey, RLookupKeyHandle, RedisModule_Free,
+    t_docId,
+};
+use rqe_iterators::metric::{MetricIterator, MetricIteratorSortedById, MetricType};
+
+#[unsafe(no_mangle)]
+/// Creates a new metric iterator sorted by ID.
+///
+/// # Safety
+///
+/// 1. `ids` must be a valid pointer to an array of `t_docId` with at least `num` elements.
+///    The array must be sorted in ascending order.
+/// 2. `metric_list` must be a valid pointer to an array of `f64` with at least `num` elements.
+/// 3. The caller must ensure that `ids` and `metric_list` are not null unless `num` is zero.
+/// 4. The memory pointed to by `ids` and `metric_list` will be freed using `RedisModule_Free`,
+///    so the caller must ensure that these pointers were allocated in a compatible manner.
+pub unsafe extern "C" fn NewMetricIteratorSortedById(
+    ids: *const t_docId,
+    metric_list: *const f64,
+    num: usize,
+    type_: MetricType,
+) -> *mut QueryIterator {
+    // SAFETY: All safety preconditions are guaranteed by the caller.
+    unsafe { new_metric_iterator::<true>(ids, metric_list, num, type_) }
```

Comment:
Not specific to this PR or function, but I was wondering... Would it not hold more value for these safety assumptions if we also couple them with `debug_asserts` the the like where possible? I think most can be done that way and for others we can probably still have checks but we possible phantom and opaque wrapper types embedded.